hudi-common/src/test/java/org/apache/hudi/model/TestHoodieAvroRecordMerger.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.model;
+
+import org.apache.hudi.common.HoodieJsonPayload;
+import org.apache.hudi.common.config.TypedProperties;
+import org.apache.hudi.common.model.HoodieAvroRecord;
+import org.apache.hudi.common.model.HoodieAvroRecordMerger;
+import org.apache.hudi.common.model.HoodieRecord;
+import org.apache.hudi.common.util.Option;
+import org.apache.hudi.common.util.collection.Pair;
+
+import org.apache.avro.Schema;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.apache.hudi.model.TestUtil.SCHEMA;
+import static org.apache.hudi.model.TestUtil.generateData;
+
+public class TestHoodieAvroRecordMerger {
```

Comment:
We should also add an functional/integration test which uses the Avro/Spark record merger to implement the same payload/merging logic as @beyond1920 mentioned and make sure that the upsert operation generates the same set of records for inserts, updates, and deletes (which is the criteria of approval on this PR), covering all new functionality of the new merge API.

Reply:
I will add that in a separate PR.